test/e2e/specs/font-size-picker.test.js

```diff
@@ -0,0 +1,105 @@
+
+		const largeButton = ( await page.$x( '//*[contains(concat(" ", @class, " "), " components-font-size-picker__buttons ")]//*[text()=\'L\']' ) )[ 0 ];
+		await largeButton.click();
+
+		// Ensure content matches snapshot.
+		const content = await getEditedPostContent();
+		expect( content ).toMatchSnapshot();
+	} );
+
+	it( 'should apply a named font size using the font size input', async () => {
+		// Create a paragraph block with some content.
+		await clickBlockAppender();
+		await page.keyboard.type( 'Paragraph to be made "small"' );
+
+		await page.click( '.blocks-font-size .components-range-control__number' );
+		await page.keyboard.type( '14' );
+
+		// Ensure content matches snapshot.
+		const content = await getEditedPostContent();
+		expect( content ).toMatchSnapshot();
+	} );
+
+	it( 'should apply a custom font size using the font size input', async () => {
+		// Create a paragraph block with some content.
+		await clickBlockAppender();
+		await page.keyboard.type( 'Paragraph to be made "small"' );
+
+		await page.click( '.blocks-font-size .components-range-control__number' );
+		await page.keyboard.type( '23' );
+
+		// Ensure content matches snapshot.
+		const content = await getEditedPostContent();
+		expect( content ).toMatchSnapshot();
+	} );
+
+	it( 'should reset a named font size using the reset button', async () => {
+		// Create a paragraph block with some content.
+		await clickBlockAppender();
+		await page.keyboard.type( 'Paragraph with font size reset using button' );
+
+		await page.click( '.blocks-font-size .components-range-control__number' );
+		await page.keyboard.type( '14' );
+
+		const resetButton = ( await page.$x( '//*[contains(concat(" ", @class, " "), " components-font-size-picker__buttons ")]//*[text()=\'Reset\']' ) )[ 0 ];
+		await resetButton.click();
+
+		// Ensure content matches snapshot.
+		const content = await getEditedPostContent();
+		expect( content ).toMatchSnapshot();
+	} );
+
+	it( 'should reset a named font size using input field', async () => {
+		// Create a paragraph block with some content.
+		await clickBlockAppender();
+		await page.keyboard.type( 'Paragraph with font size reset using input field' );
+
+		const largeButton = ( await page.$x( '//*[contains(concat(" ", @class, " "), " components-font-size-picker__buttons ")]//*[text()=\'L\']' ) )[ 0 ];
+		await largeButton.click();
+
+		await page.click( '.blocks-font-size .components-range-control__number' );
+		await page.keyboard.press( 'Backspace' );
+		await page.keyboard.press( 'Backspace' );
+
+		// Ensure content matches snapshot.
+		const content = await getEditedPostContent();
+		expect( content ).toMatchSnapshot();
+	} );
+
+	it( 'should reset a custom font size using input field', async () => {
+		// Create a paragraph block with some content.
+		await clickBlockAppender();
+		await page.keyboard.type( 'Paragraph to be made "small"' );
+
+		await page.click( '.blocks-font-size .components-range-control__number' );
```

Comment:
if the logic between lines 92 and 100 won't work, this test will still pass. The same applies to other tests which reset the font size. Do you have an idea how to mitigate it?

Reply:
Hi @gziolo to mitigate this problem, we have very similar tests containing the part where we set the font size. So if lines 92-100 don't work this test will pass but the tests where we set the font size are going to fail because the logic is the same. In summary, if lines 92-100 don't work it means the setting of the font size is not working and we are testing that.
Another approach would be to use the data-layer functions to directly insert the block with the font size already set. But even this approach is not perfect if the block was not correctly created the test may still pass wrongly.

Reply:
Sounds reasonable 👍 I wasn’t aware that it is tested in other place